pip/pep425tags.py

```diff
@@ -113,6 +113,9 @@ def get_supported(versions=None, noarch=False):
             for arch in arches:
                 supported.append(('%s%s' % (impl, versions[0]), abi, arch))
 
+        # Has binaries, does not use the Python API:
+        supported.append(('py%s' % (versions[0][0]), 'none', arch))
```

Comment:
So this code depends on the versions[0][0] input and on the arch... whats the right behaviour when noarch is false? A test for it might be to assert that
get_supported([['2']], False) results in ('py2', 'none', local_arch) in the output, and for get_support([['2]], True) to do ??? [whats the right thing here?].

We could make this more meaningfully testable by pushing all the calculation to pure functions, and having get_supported just do the querying and then call those functions